src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt

```diff
@@ -122,11 +130,10 @@ class VisitService(
     visitDto: CreateVisitRequest,
     offenderBooking: OffenderBooking,
   ) {
-    val offenderVisitBalance = offenderVisitBalanceRepository.findById(offenderBooking.bookingId)
+    val offenderVisitBalance = offenderBooking.visitBalance
+    if (offenderVisitBalance != null) {
 
-    if (offenderVisitBalance.isPresent) {
-
-      if (offenderVisitBalance.get().remainingPrivilegedVisitOrders!! > 0) {
+      if (offenderVisitBalance.remainingPrivilegedVisitOrders!! > 0) {
```

Comment:
very minor, but pretty certain you don't need `!!` here due to line 134 - the Kotlin complier is clever like that

Reply:
yep youre right!